src/main/java/org/variantsync/diffdetective/variationtree/VariationTreeNode.java

```diff
@@ -0,0 +1,364 @@
+    }
+
+    @Override
+    public List<VariationNode<VariationTreeNode>> getChildren() {
+        return Collections.unmodifiableList(childOrder);
+    }
+
+    @Override
+    public void addChild(final VariationNode<VariationTreeNode> child) {
+        child.getBackingNode().setParent(this);
+        childOrder.add(child.getBackingNode());
+    }
+
+    @Override
+    public void addChild(final VariationNode<VariationTreeNode> child, int index) {
+        child.getBackingNode().setParent(this);
+        childOrder.add(index, child.getBackingNode());
+    }
+
+    @Override
+    public boolean removeChild(final VariationNode<VariationTreeNode> child) {
+        if (isChild(child)) {
+            child.getBackingNode().parent = null;
+            childOrder.remove(child);
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void removeAllChildren() {
+        for (var child : childOrder) {
+            child.getBackingNode().parent = null;
+        }
+
+        childOrder.clear();
+    }
+
+    @Override
+    public Node getDirectFeatureMapping() {
+        return featureMapping;
+    }
+
+    /**
+     * Returns an integer that uniquely identifiers this node within its variation tree.
+     *
+     * <p>From the returned id a new node with all essential attributes ({@link getNodeType node
+     * type} and {@link getFromLine start line number}) can be reconstructed by using
+     * {@link DiffNode#fromID}.
+     *
+     * <p>Note that this encoding assumes that line numbers fit into {@code 26} bits.
+     */
+    @Override
+    public int getID() {
+        // Add one to ensure invalid (negative) line numbers don't cause issues.
+        int lineNumber = 1 + from;
+        Assert.assertTrue((lineNumber << 2*ID_OFFSET) >> 2*ID_OFFSET == lineNumber);
+
+        int id;
+        id = lineNumber;
+        id <<= ID_OFFSET;
+        // This makes `VariationTreeNode.toID` compatible with `DiffNode.toID`
+        id += DiffType.NON.ordinal();
+        id <<= ID_OFFSET;
+        id += nodeType.ordinal();
+        return id;
+    }
+
+    /**
+     * Reconstructs a node from the given {@link getID id} and label.
+     * The almost-inverse function is {@link DiffNode#getID()} but the conversion is not lossless.
+     *
+     * @param id the id from which to reconstruct the node
+     * @param label the label the node should have
+     * @return the reconstructed node
+     */
+    public static VariationTreeNode fromID(final int id, List<String> label) {
+        final int lowestBitsMask = (1 << ID_OFFSET) - 1;
+
+        final int nodeTypeOrdinal = id & lowestBitsMask;
+        Assert.assertTrue(((id >> ID_OFFSET) & lowestBitsMask) == DiffType.NON.ordinal());
+        final int from = (id >> (2*ID_OFFSET)) - 1;
+
+        return new VariationTreeNode(
+                NodeType.values()[nodeTypeOrdinal],
+                from,
+                DiffLineNumber.InvalidLineNumber,
+                null,
```

Comment:
Use true if we have an annotation node? Maybe its best to keep null here though as a "reminder" that this node is still incomplete.

Reply:
:facepalm: This is the exact same bug as I fixed in #50 (it caused the test failures). This needs to be fixed otherwise `assertConsistency` will detect that and we would have an unexpected inconsistent behavior between `DiffTreen.assertConsistency` and `VariationNode.assertConsistency`.

Reply:
Maybe we should document in the method's javadoc which information is lost and only recovered by heuristics (i.e., line numbers and formula of annotations)?

Reply:
I don't think it's a good idea to give definite answers to what exactly can be recovered from this ID because this would encourage using it for serialization. Misusing `getID` for serializing some fields of `DiffNode` was a mistake in my opinion. It's only real use case should be identifying a node uniquely in a tree.
It's probably a good idea to completely eliminate `fromID` in the future as it's not really stable across versions (I already changed the IDs multiple times) and implicitly looses information.

Reply:
I see your point. So the misuse would be any use of `fromID` but `toID` should be fine. I implemented these methods because of a use case were we had to export DiffTrees for another tool that reads linegraph files which in turn require natural number identifiers for nodes. I agree that upon serialisation we should either use a proper format that stores node data separately (e.g., in a json or in a text label in the linegraph file).